### PR TITLE
Only run the sbt scripted tests and CB when publishing sbt-dotty

### PR DESCRIPTION
**Pull request by smarter**
Everything else is independent of sbt-dotty working or not.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -210,11 +218,12 @@ jobs:
   test_java8:
     runs-on: [self-hosted, Linux]
     container: lampepfl/dotty:2020-04-24
-    if: (
+    if: "(
```
Comment (anatoliykmetyuk) [inline .github/workflows/ci.yaml:221]: Why this change?

**Review by anatoliykmetyuk — APPROVED** [review]
Makes sense.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (smarter) [reply .github/workflows/ci.yaml:221]: Quotes are needed when using ! apparently
